src/jaegertracing/propagation/Propagator.h

```diff
@@ -62,14 +62,17 @@ class Propagator : public Extractor<ReaderType>, public Injector<WriterType> {
         SpanContext ctx;
         StrMap baggage;
         std::string debugID;
+        std::string traceState;
         const auto result = reader.ForeachKey(
-            [this, &ctx, &debugID, &baggage](const std::string& rawKey,
-                                             const std::string& value) {
+            [this, &ctx, &debugID, &baggage, &traceState](
+                const std::string& rawKey, const std::string& value) {
                 const auto key = normalizeKey(rawKey);
                 if (key == _headerKeys.traceContextHeaderName()) {
```

Comment:
could we split the default Jaeger propagation header handling from `traceparent`?  I.e. do not overload overridable `traceContextHeaderName()` use case with W3C propagation at all.

In all other Jaeger SDKs, the default Jaeger propagator and W3C propagator are implemented as independent classes.

Reply:
I will have a look.

To be sure, if a user selects the w3c format but not jaeger format, should only the two w3c headers be supported or also e.g. jaeger-debug-id?

Reply:
It's kind of up to you. Strictly speaking, the two extra headers `jaeger-debug-id` and `jaeger-baggage` are features of the default Jaeger propagator, so if you instead configure the B3 propagator they might be either supported on not, depending on the Jaeger client implementation. But I think it's a nice-to-have to support `jaeger-debug-id` in the W3C propagator since it provides an additional debugging capability to the users.